x-pack/filebeat/input/httpjson/request.go

```diff
@@ -154,38 +178,185 @@ func (rf *requestFactory) newHTTPRequest(stdCtx context.Context, trCtx *transfor
+			if err != nil {
+				return err
+			}
+			n, err = r.processAndPublishEvents(stdCtx, trCtx, publisher, intermediateResps, false)
+			if err != nil {
+				return err
+			}
+		} else {
+			if len(ids) == 0 {
+				n = 0
+				continue
+			}
+			urlCopy = rf.url
+			urlString = rf.url.String()
+			// perform request over collected ids
+			for _, id := range ids {
+				// reformat urls of requestFactory using ids
+				rf.url, err = generateNewUrl(rf.replace, urlString, id)
+				if err != nil {
+					return fmt.Errorf("failed to generate new URL: %w", err)
+				}
+
+				// collect data from new urls
+				httpResp, err = rf.collectResponse(stdCtx, trCtx, r)
+				if err != nil {
+					return fmt.Errorf("failed to execute rf.collectResponse: %w", err)
+				}
+				// store data according to response type
+				if i == len(r.requestFactories)-1 && len(ids) != 0 {
+					finalResps = append(finalResps, httpResp)
+				} else {
+					intermediateResps = append(intermediateResps, httpResp)
+				}
+			}
+			rf.url = urlCopy
+
+			var resps []*http.Response
+			if i < len(r.requestFactories) {
+				resps = finalResps
+			} else {
+				// The if comdition (i < len(r.requestFactories)) ensures this branch never runs to the last element
+				// of r.requestFactories, therefore r.requestFactories[i+1] will never be out of bounds.
+				ids, err = r.getIdsFromResponses(intermediateResps, r.requestFactories[i+1].replace)
+				if err != nil {
+					return err
+				}
+				resps = intermediateResps
+			}
+			split = r.responseProcessor.split
+			r.responseProcessor.split = rf.split
+			n, err = r.processAndPublishEvents(stdCtx, trCtx, publisher, resps, i < len(r.requestFactories))
+			if err != nil {
+				return err
+			}
+			r.responseProcessor.split = split
+		}
+	}
+
+	r.log.Infof("request finished: %d events published", n)
+
+	return nil
+}
+
+// getIdsFromResponses returns ids from responses
+func (r *requester) getIdsFromResponses(intermediateResps []*http.Response, replace string) ([]string, error) {
+	var b []byte
+	var ids []string
+	var err error
+	// collect ids from all responses
+	for _, resp := range intermediateResps {
+		if resp.Body != nil {
+			b, err = io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("error while reading response body: %w", err)
+			}
+		}
+
+		// get replace values from collected json
+		var v interface{}
+		json.Unmarshal(b, &v)
+		values, err := jsonpath.Get(replace, v)
+		if err != nil {
+			return nil, fmt.Errorf("error while getting keys: %w", err)
+		}
+
+		switch tresp := values.(type) {
+		case []interface{}:
+			for _, v := range tresp {
+				_, ok := v.(map[string]interface{})
+				if ok {
+					r.log.Debugf("events must be int, string or double, but got %T: skipping", v)
+					continue
+				}
+				ids = append(ids, fmt.Sprintf("%v", v))
+			}
+		case map[string]interface{}:
+			r.log.Debugf("cannot collect IDs from type '%T' : '%v'", values, values)
```

Comment:
Those logs are showing errors that will affect the final result, they should be at least level warning, maybe even error
```suggestion
					r.log.Errorf("events must be int, string or double, but got %T: skipping", v)
					continue
				}
				ids = append(ids, fmt.Sprintf("%v", v))
			}
		case map[string]interface{}:
			r.log.Errorf("cannot collect IDs from type '%T' : '%v'", values, values)
```

Reply:
Got it.